gustaf/io/npz.py

```diff
@@ -0,0 +1,162 @@
+"""gustaf/gustaf/io/npz.py
+
+io functions for numpy.savez.
+"""
+
+import numpy as np
+
+from gustaf import utils
+from gustaf.vertices import Vertices
+from gustaf.edges import Edges
+from gustaf.faces import Faces
+from gustaf.volumes import Volumes
+
+_stored_arrays = {
+        Vertices: ('vertices',),
+        Edges: ('vertices', 'edges'),
+        Faces: ('vertices', 'faces'),
+        Volumes: ('vertices', 'volumes')
+        }
+
+_stored_dicts = {
+        Vertices: (),
+        Edges: (),
+        Faces: ('BC',),
+        Volumes: ('BC',)
+        }
+
+def load(
+        fname,
+        force_type = None,
+        check = False
+):
+    """
+    Load an npz file.
+
+    The type is determined from the 'kind' array. If force_type is used, this
+    type must match what is found in 'kind'.
+
+    Parameters
+    -----------
+    fname : str
+    force_type : type or None
+        (Default: None)
+    check : bool
+        (Default: False) Check consistency of connectivity data.
+
+    Returns
+    -------
+    mesh : Volumes, Faces, Edges, or Vertices
+    """
+    # read a dictionary from the file
+    with np.load(fname, allow_pickle=False) as data:
+        # find kind and set mesh class type
+        try:
+            mesh_kind = data['kind'][0]
+        except KeyError:
+            raise RuntimeError("NPZ file is missing 'kind' field. Found "
+                    + ", ".join(data) + ".")
+
+        # prepare kind => type dict (e.g. 'volume': Volumes)
+        kind_to_type = {type_.kind: type_ for type_ in _stored_arrays.keys()}
+        try:
+            mesh_type = kind_to_type[mesh_kind]
+        except KeyError:
+            raise RuntimeError(f"Kind '{mesh_kind}' doesn't match any known "
+                    + "types: " + ", ".join(kind_to_type) + ".")
+
+        if force_type is not None and force_type != mesh_type:
+            raise RuntimeError(f"Requested {force_type} but got {mesh_type}.")
+
+        # loop over all array entries and assign correctly
+        # (we're importing into immediate variables so we can make sure all
+        # arrays are imported before the dicts.)
+        read_arrays = {}
+        read_dicts = {}
+        for name, content in data.items():
+            # check for array
+            if name in _stored_arrays[mesh_type]:
+                read_arrays[name] = content
+            elif name == "kind":
+                continue
+            else:
+                dict_name, *dict_entry_name = name.split("_", 1)
+                # check for dict
+                if dict_name in _stored_dicts[mesh_type]:
+                    if dict_name not in read_dicts:
+                        read_dicts[dict_name] = {}
+                    read_dicts[dict_name][dict_entry_name[0]] = content
+                else:
+                    utils.log.warning("Encountered unknown array "
+                            f"'{name}' while reading NPZ file.")
```

Comment:
```suggestion
                # unpacking nested keys based on naming convention.
                # see `export()`'s doc
                dict_name, dict_entry_name = name.split("_", 1)
                # check for dict
                if dict_name in _stored_dicts[mesh_type]:
                    if dict_name not in read_dicts:
                        read_dicts[dict_name] = {}
                    read_dicts[dict_name][dict_entry_name] = content
                else:
                    utils.log.warning("Encountered unknown array "
                            f"'{name}' while reading NPZ file.")
```

Reply:
You've removed the asterisk in your suggestion but that cannot be done. It leads to a `ValueError` if `name` doesn't contain an underscore.